pkg/collect/result.go

```diff
@@ -142,21 +142,18 @@ func (r CollectorResult) ReplaceResult(bundlePath string, relativePath string, r
 		return nil
 	}
 
-	tmpFile, err := os.CreateTemp("", "replace-")
-	if err != nil {
-		return errors.Wrap(err, "failed to create temp file")
-	}
-	defer tmpFile.Close()
+	targetPath := filepath.Join(bundlePath, relativePath)
```

Comment:
curious re the old vs new approaches to handling this:

- the old creates a file in `/tmp`, then writes content to it, then renames it (in the same partition supposedly)
- the new creates the desired filename instead, and writes to that directly

is there a noticeable IO improvement to the new...? In theory both approaches write to the partition once, and the rename operation should be very fast...?

Not opposed to the change in behaviour, just curious about the before/after benchmarks :) I know there were issues around this operation in the past being slow.

Reply:
Very good point, after testing, none of the files were moving between different partitions. And if we write to the same file, the previous io.read will conflict with io.Copy which causes less file size to be redact. 
For example, we have `apiserver-audit-logs/k8s-audit-2023-08-08T01-47-13.742.log` with origin size `100M`.
Without writing to a tmp file and rename, the result file size is 4K

That's why it feels like we have reduce the redact time, actually, it is a mistake.

So currently, I switch the current function to use goroutine to reduce waiting time of redacting each files.

It can reduce the time 
```
Duration: 230,272ms                 ->                188,182ms
```